tdvt/tdvt/config_gen/test_config.py

```diff
@@ -62,24 +65,20 @@ def generate_test_file_list(self):
            Return the sorted list of tests.
 
         """
-        final_test_list = self.generate_test_file_list_from_config()
+        if self.test_list_checked:
+            return self.test_list_cached
 
-        logging.debug("Found final list of " + str(len(final_test_list)) + " tests to run.")
-        if len(final_test_list) == 0:
-            logging.warning("Did not find any tests to run.")
-            print("Did not find any tests to run.")
-            return final_test_list
+        final_test_list = self.__generate_test_file_list()
 
-        for x in final_test_list:
-            logging.debug("final test path " + x.test_path)
-
-        return final_test_list
+        self.test_list_cached = final_test_list
+        self.test_list_checked = True
+        return self.test_list_cached
 
     def get_test_dirs(self):
         return (self.root_dir, get_local_test_dir())
 
-    def generate_test_file_list_from_config(self):
-        """Read the config file and generate a list of tests."""
+    def __generate_test_file_list(self):
```

Comment:
Since we've got a few methods called `generate_test_file_list` now, it'd be great if we could add type hints to them.